provisioner/clusterpy.go

```diff
@@ -310,11 +311,6 @@ func (p *clusterpyProvisioner) Provision(ctx context.Context, logger *log.Entry,
 		return err
 	}
 
-	instanceTypes, err := awsUtils.NewInstanceTypesFromAWS(awsAdapter.ec2Client)
```

Comment:
This I moved into the `prepareProvision` function. The data is used in that function and returned as well so it can be used in `Provision` function as well. This way we fetch the data just once on every call to `Provision`.